actors/builtin/multisig/multisig_test.go

```diff
@@ -311,6 +311,519 @@ func TestApprove(t *testing.T) {
+		rt.Verify()
+
+		// the number of signers has decreased, specifically chuck is out.
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		assert.Equal(t, initialSigner[:2], st.Signers)
+		assert.Equal(t, numApprovals, st.NumApprovalsThreshold)
+		assert.Equal(t, abi.NewTokenAmount(0), st.InitialBalance)
+		assert.Equal(t, abi.ChainEpoch(0), st.UnlockDuration)
+		assert.Equal(t, abi.ChainEpoch(0), st.StartEpoch)
+	})
+
+	t.Run("remove signer with manual threshold decrease", func(t *testing.T) {
+		const numApprovals = int64(2)
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, initialSigner...)
+
+		rt.SetCaller(receiver, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerAddr(receiver)
+		actor.removeSigner(rt, initialSigner[2], true)
+		rt.Verify()
+
+		// the number of approvals in additions to the number of signers has decreased, chuck is gone, sorry chuck.
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		assert.Equal(t, initialSigner[:2], st.Signers)
+		assert.Equal(t, numApprovals-1, st.NumApprovalsThreshold)
+		assert.Equal(t, abi.NewTokenAmount(0), st.InitialBalance)
+		assert.Equal(t, abi.ChainEpoch(0), st.UnlockDuration)
+		assert.Equal(t, abi.ChainEpoch(0), st.StartEpoch)
+	})
+
+	t.Run("remove signer with automatic threshold decrease to avoid lockout", func(t *testing.T) {
+		const numApprovals = int64(3)
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, initialSigner...)
+
+		rt.SetCaller(receiver, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerAddr(receiver)
+		actor.removeSigner(rt, initialSigner[2], false)
+		rt.Verify()
+
+		// the number of approvals in additions to the number of signers has decreased, chuck is gone, sorry chuck.
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		assert.Equal(t, initialSigner[:2], st.Signers)
+		assert.Equal(t, numApprovals-1, st.NumApprovalsThreshold)
+		assert.Equal(t, abi.NewTokenAmount(0), st.InitialBalance)
+		assert.Equal(t, abi.ChainEpoch(0), st.UnlockDuration)
+		assert.Equal(t, abi.ChainEpoch(0), st.StartEpoch)
+	})
+
+	t.Run("unauthorized signer fails to remove signer", func(t *testing.T) {
+		richard := newIDAddr(t, 104)
+
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, initialSigner...)
+
+		rt.SetCaller(receiver, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerAddr(receiver)
+		rt.ExpectAbort(exitcode.ErrPlaceholder, func() {
+			actor.removeSigner(rt, richard, false)
+		})
+		rt.Verify()
+
+		// the signers remain unchanged after a failed attempt to remove.
+		var st multisig.MultiSigActorState
+		rt.Readonly(&st)
+		assert.Equal(t, initialSigner, st.Signers)
+		assert.Equal(t, numApprovals, st.NumApprovalsThreshold)
+		assert.Equal(t, abi.NewTokenAmount(0), st.InitialBalance)
+		assert.Equal(t, abi.ChainEpoch(0), st.UnlockDuration)
+		assert.Equal(t, abi.ChainEpoch(0), st.StartEpoch)
+	})
```

Comment:
Test failure to remove last signer (or success, if that's what it does)

Similar to above, I would collapse these into less code with more re-use of intermediate state.